File: orbps-web-app/src/main/webapp/orbps/otherFunction/batchAppManage/js/batchControl.js

$(function() {
	//批作业批量启动
	$("#taskRegStartBtn").click(function(){
		lion.util.info("批作业批量启动中,请稍候。。。");
		lion.util.postjson("/orbps/web/orbps/otherFunction/branchControl/batchRestart",null,com.orbps.otherFunction.bacthAppManage.batchStart);
	});
	com.orbps.otherFunction.bacthAppManage.batchStart = function(data,args){
		if(lion.util.isEmpty(data)){
			lion.util.info("批作业启动失败");
			return false;
		}
		if("0" === data.errCode){
			lion.util.info(data.errMsg);
		}
	}
	
	//批作业批量停止
	$("#taskRegStopBtn").click(function(){
		lion.util.info("批作业批量停止中,请稍候。。。");
		lion.util.postjson("/orbps/web/orbps/otherFunction/branchControl/batchStop", null,com.orbps.otherFunction.bacthAppManage.batchStop);
	});
	
	com.orbps.otherFunction.bacthAppManage.batchStop = function(data,args){
		if(lion.util.isEmpty(data)){
			lion.util.info("批作业停止失败");
			return false;
		}
		if("0" === data.errCode){
			lion.util.info(data.errMsg);
		}
	}
	
	//批作业启停结果查询
	$("#batchStatQueryBtn").click(function(){
		lion.util.postjson("/orbps/web/orbps/otherFunction/branchControl/batchStatQuery", null,com.orbps.otherFunction.bacthAppManage.batchQuery);
	});

	com.orbps.otherFunction.bacthAppManage.batchQuery = function(data,args){
		if(lion.util.isEmpty(data)){
			lion.util.info("批作业停止状态查询失败");
			return false;
		}
		
		lion.util.info(data.errMsg);
	}
});

File: orbps-web-app/src/main/java/com/newcore/orbps/web/otherfunction/BatchControl.java
package com.newcore.orbps.web.otherfunction;
import org.apache.shiro.session.Session;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import com.halo.core.header.HeaderInfoHolder;
import com.halo.core.web.annotation.CurrentSession;
import com.halo.core.web.annotation.ResponseMessage;
import com.newcore.orbps.models.para.RetInfo;
import com.newcore.orbps.service.api.BatchJobControlService;
import com.newcore.orbps.web.contractentry.OfflineListImportController;
import com.newcore.supports.models.cxf.CxfHeader;
import com.newcore.supports.util.HeaderInfoUtils;

/**
 * 机构权限控制
 * 
 * @author 靳聪
 *
 */
@Controller
@RequestMapping("/orbps/otherFunction/branchControl")
public class BatchControl {
	/**
     * 日志对象.
     */
    private static Logger logger = LoggerFactory.getLogger(OfflineListImportController.class);
	
	/**
     * 批作业控制服务
     */
    @Autowired
    BatchJobControlService batchJobControlService;
	
    /**
     * 批量重启批作业服务
     * 
     * @author jincong
     * @param session
     * @param str
     * @return
     */
    @RequestMapping(value = "/batchRestart")
    public @ResponseMessage RetInfo batchRestart(@CurrentSession Session session) {
    	RetInfo retInfo = new RetInfo();
	    try{
	    	CxfHeader headerInfo = HeaderInfoUtils.buildDefaultHeaderInfo();
	        HeaderInfoHolder.setOutboundHeader(headerInfo);
	    	batchJobControlService.restartStoppedBatJob();
	    }catch(Exception e){
	        logger.info(e.getMessage(), e);
	        retInfo.setRetCode("0");
	        retInfo.setErrMsg("批作业启动失败！");
	    }
        return retInfo;
    }
    
    /**
     * 批量停止批作业服务
     * 
     * @author jincong
     * @param session
     * @param str
     * @return
     */
    @RequestMapping(value = "/batchStop")
    public @ResponseMessage RetInfo batchStop(@CurrentSession Session session) {
    	RetInfo retInfo = new RetInfo();
	    try{
	    	CxfHeader headerInfo = HeaderInfoUtils.buildDefaultHeaderInfo();
	        HeaderInfoHolder.setOutboundHeader(headerInfo);
	    	batchJobControlService.stopMultiBatchJob();
	    }catch(Exception e){
	        logger.info(e.getMessage(), e);
	        retInfo.setRetCode("0");
	        retInfo.setErrMsg("批作业停止失败！");
	    }
        return retInfo;
    }
    
    /**
     * 批作业启停结果查询
     * 
     * @author jincong
     * @param session
     * @param str
     * @return
     */
    @RequestMapping(value = "/batchStatQuery")
    public @ResponseMessage RetInfo batchStatQuery(@CurrentSession Session sessionss) {
    	RetInfo retInfo = new RetInfo();
    	CxfHeader headerInfo = HeaderInfoUtils.buildDefaultHeaderInfo();
        HeaderInfoHolder.setOutboundHeader(headerInfo);
	    try{
	    	retInfo = batchJobControlService.getIsBatchStopped();
	    }catch(Exception e){
	        logger.info(e.getMessage(), e);
	        retInfo.setRetCode("0");
	        retInfo.setErrMsg("批作业查询失败！");
	    }
        return retInfo;
    }
}